I have added some additional checks. It seems the tests are incorrect, or I got the checks wrong:

```
--- FAIL: TestGoKZGVerifyPartialCells (0.48s)
    kzg4844_test.go:305: iter 0: failed to verify partial cells: less cellIndices than cells
--- FAIL: TestGoKZGRecoverBlob (0.48s)
    kzg4844_test.go:396: iter 0: failed to recover blob with 118 cells: less cellIndices than cells
```

--- a/crypto/kzg4844/kzg4844.go
+++ b/crypto/kzg4844/kzg4844.go
@@ -212,7 +212,6 @@ func IsValidVersionedHash(h []byte) bool {
 // VerifyCells verifies a batch of proofs corresponding to the cells and blob commitments.
 //
 // For this function, it is sufficient to only provide some of the cells.
-// For each blob being proven, the cells slice must contain at least 64 items.
 //
 // The `cellIndices` specify which of the 128 cells of each blob are given.
 // Note the list of indices is shared among all blobs, i.e. for a given list of indices
@@ -249,6 +248,7 @@ func ComputeCells(blobs []Blob) ([]Cell, error) {
 }
 
 // RecoverBlobs recovers blobs from the given cells and cell indices.
+// In order to successfully recover, at least DataPerBlob (64) cells must be provided.
 //
 // For the layout of cells and cellIndices, please see [VerifyCells].
 func RecoverBlobs(cells []Cell, cellIndices []uint64) ([]Blob, error) {

--- a/crypto/kzg4844/kzg4844.go
+++ b/crypto/kzg4844/kzg4844.go
@@ -215,8 +215,9 @@ func IsValidVersionedHash(h []byte) bool {
 // For each blob being proven, the cells slice must contain at least 64 items.
 //
 // The `cellIndices` specify which of the 128 cells of each blob are given.
-// Thus, `len(cellIndices)` must be >= 64 and <= 128 to be valid.
-// `len(cells)` must be a multiple of `len(cellIndices)`.
+// Note the list of indices is shared among all blobs, i.e. for a given list of indices
+// [1, 2, 13], the cells slice must contain cells [1, 2, 13] of each blob.
+// Thus, `len(cells)` must be a multiple of `len(cellIndices)`.
 //
 // One proof must be given for each cell. As such, `len(proofs)` must equal `len(cells)`.
 func VerifyCells(cells []Cell, commitments []Commitment, proofs []Proof, cellIndices []uint64) error {

--- a/crypto/kzg4844/kzg4844_gokzg.go
+++ b/crypto/kzg4844/kzg4844_gokzg.go
@@ -18,7 +18,6 @@ package kzg4844
 
 import (
 	"encoding/json"
-	"errors"
 	"sync"
 
 	gokzg4844 "github.com/crate-crypto/go-eth-kzg"
@@ -203,10 +202,6 @@ func gokzgComputeCells(blobs []Blob) ([]Cell, error) {
 // gokzgRecoverBlobs recovers blobs from cells and cell indices.
 func gokzgRecoverBlobs(cells []Cell, cellIndices []uint64) ([]Blob, error) {
 	gokzgIniter.Do(gokzgInit)
-
-	if len(cellIndices) == 0 || len(cells)%len(cellIndices) != 0 {
-		return []Blob{}, errors.New("cells with wrong length")
-	}
 
 	blobCount := len(cells) / len(cellIndices)
 	blobs := make([]Blob, 0, blobCount)

